apps/meteor/ee/app/livechat-enterprise/server/hooks/resumeOnHold.ts

```diff
@@ -1,46 +1,73 @@
+import { LivechatRooms, LivechatVisitors, Users } from '@rocket.chat/models';
+import { TAPi18n } from 'meteor/rocketchat:tap-i18n';
+import { OmnichannelEEService } from '@rocket.chat/core-services';
 
 import { callbacks } from '../../../../../lib/callbacks';
+import { callbackLogger } from '../../../../../app/livechat/server/lib/callbackLogger';
 
-callbacks.add(
-	'afterSaveMessage',
-	async (message, roomParams) => {
-		// skips this callback if the message was edited
-		if (isEditedMessage(message)) {
-			return message;
-		}
+const resumeOnHoldCommentAndUser = async (room: IOmnichannelRoom): Promise<{ comment: string; resumedBy: IUser }> => {
+	const {
+		v: { _id: visitorId },
+		_id: rid,
+	} = room;
+	const visitor = await LivechatVisitors.findOneById<Pick<ILivechatVisitor, 'name' | 'username'>>(visitorId, {
+		projection: { name: 1, username: 1 },
+	});
+	if (!visitor) {
+		callbackLogger.error(`[afterSaveMessage] Visitor Not found for room ${rid} while trying to resume on hold`);
+		throw new Error('Visitor not found while trying to resume on hold');
+	}
 
-		// if the message has a type means it is a special message (like the closing comment), so skips
-		if (message.t) {
-			return message;
-		}
+	const guest = visitor.name || visitor.username;
 
-		if (!isOmnichannelRoom(roomParams)) {
-			return message;
-		}
+	const resumeChatComment = TAPi18n.__('Omnichannel_on_hold_chat_automatically', { guest });
 
-		const { _id: rid, t: roomType, v: roomVisitor } = roomParams;
+	const resumedBy = await Users.findOneById('rocket.cat');
+	if (!resumedBy) {
+		callbackLogger.error(`[afterSaveMessage] User Not found for room ${rid} while trying to resume on hold`);
+		throw new Error(`User not found while trying to resume on hold`);
+	}
 
-		// message valid only if it is a livechat room
-		if (!(typeof roomType !== 'undefined' && roomType === 'l' && roomVisitor && roomVisitor.token)) {
-			return message;
-		}
+	return { comment: resumeChatComment, resumedBy };
+};
+
+const handleAfterSaveMessage = async (message: IMessage, room: IRoom) => {
+	if (isEditedMessage(message) || message.t || !isOmnichannelRoom(room)) {
+		return message;
+	}
+
+	const { _id: rid, v: roomVisitor } = room;
+
+	if (!roomVisitor?._id) {
+		return message;
+	}
+
+	// Need to read the room every time, the room object is not updated
+	const updatedRoom = await LivechatRooms.findOneById(rid);
+	if (!updatedRoom) {
+		return message;
+	}
 
-		// Need to read the room every time, the room object is not updated
-		const room = await LivechatRooms.findOneById(rid, { projection: { t: 1, v: 1, onHold: 1 } });
-		if (!room) {
+	if (message.token && room.onHold) {
+		callbackLogger.debug(`[afterSaveMessage] Room ${rid} is on hold, resuming it now since visitor sent a message`);
+
+		try {
+			const { comment: resumeChatComment, resumedBy } = await resumeOnHoldCommentAndUser(updatedRoom);
+			await OmnichannelEEService.resumeRoomOnHold(updatedRoom, resumeChatComment, resumedBy);
```

Comment:
I don't think that would be a good idea and we're not doing this right now. Here on line 57, we're only resolving the comment and commentor. We're not sending that message. Rather this info is being passed to the resumeOnHold service which would use this info to send the system message once the chat gets resumed.

Sidenote: We're passing this value as a param since chat can be resumed in 2 ways:
1) First by a visitor when they send a message
2) Or by an agent when they click on "Resume" button on the UI